src/rgw/rgw_data_sync.cc

```diff
@@ -4514,6 +4516,20 @@ int RGWBucketShardIncrementalSyncCR::operate(const DoutPrefixProvider *dpp)
       }
     }
 
+    yield {
```

Comment:
@cbodley I'm not sure if this is the right place to add shard object removal. We could add this piece after we finish running `sync_bucket_shard_cr` in `RGWRunBucketSourcesSyncCR()` to make sure that the sync is complete for a particular shard before removal.